control-plane/rest/openapi-specs/v0_api_spec.yaml

```diff
@@ -2375,6 +2399,117 @@ components:
+        - Init
+        - Rebuilding
+        - Stopped
+        - Paused
+        - Failed
+        - Completed
+    RebuildHistory:
+      example:
+        - childUri: 'nvmf://10.1.0.5:8420/nqn.2019-05.io.openebs:37d83441-e8ef-4e17-a29e-25169d91cb96'
+          srcUri: 'nvmf://10.1.0.9:8420/nqn.2019-05.io.openebs:37d83441-e8ef-4e17-a29e-25169d91cb25'
+          state: Completed
+          blocksTotal: 0
+          blocksRecovered: 0
+          blocksTransferred: 0
+          blocksRemaining: 0
+          blocksPerTask: 0
+          blockSize: 0
+          isPartial: true
+          startTime: ''
+          endTime: ''
+      description: Rebuild History of a Child
+      type: object
+      properties:
+        childUri:
+          description: Uri of the rebuilding child
+          type: string
+        srcUri:
+          description: Uri of source child for rebuild job
+          type: string
+        rebuildJobState:
+          $ref: '#/components/schemas/RebuildJobState'
+        blocksTotal:
+          description: Total blocks to rebuild
+          type: integer
+          format: uint64
+        blocksRecovered:
+          description: Number of blocks processed
+          type: integer
+          format: uint64
+        blocksTransferred:
+          description: Number of blocks to transferred
+          type: integer
+          format: uint64
+        blocksRemaining:
+          description: Number of blocks remaining
+          type: integer
+          format: uint64
+        blocksPerTask:
+          description: Total blocks to per task in the job
+          type: integer
+          format: uint64
+        blockSize:
+          description: Size of each block in the task
+          type: integer
+          format: uint64
+        isPartial:
+          description: True means its Partial rebuild job. If false, its Full rebuild job
+          type: boolean
+        startTime:
+          description: Start time of the rebuild job (UTC)
+          type: string
+          format: date-time
+        endTime:
+          description: End time of the rebuild job (UTC)
+          type: string
+          format: date-time
+      required:
+        - childUri
+        - srcUri
+        - rebuildJobState
+        - blocksTotal
+        - blocksRecovered
+        - blocksTransferred
+        - blockRemaining
+        - blocksPerTask
+        - blockSize
+        - isPartial
+        - startTime
+        - endTime
+    RebuildRecord:
+      example:
+        nexusId: 514ed1c8-7174-49ac-b9cd-ad44ef670a67
+        histories:
+          - childUri: 'nvmf://10.1.0.5:8420/nqn.2019-05.io.openebs:37d83441-e8ef-4e17-a29e-25169d91cb96'
+            srcUri: 'nvmf://10.1.0.9:8420/nqn.2019-05.io.openebs:37d83441-e8ef-4e17-a29e-25169d91cb25'
+            state: Completed
+            blocksTotal: 0
+            blocksRecovered: 0
+            blocksTransferred: 0
+            blocksRemaining: 0
+            blocksPerTask: 0
+            blockSize: 0
```

Comment:
Do we really need to expose all these fields via REST? I guess only a subset should be sufficient. e.g. blocksPerTask is a typical internal detail.

Reply:
I think we can skip this field.

Reply:
Removed blocks_per_task. Others seem okay to expose.